### ISSUE TITLE
Documentation for registry Compat.toml version conventions?

### ISSUE THREAD

**Issue by GordStephen** [JuliaLang/Pkg.jl#1190]
Does any documentation exist describing how versions should be specified in Compat.toml files in a registry? My initial assumption was that it would be the same rules [as documented for package-level compat specifications](https://julialang.github.io/Pkg.jl/v1/compatibility/), but that doesn't seem to be the case.

Are the options just `exact.version.number` or `min.version.number-max.version.number`? I see a lot of `x.y.z-0` in General, is that a way to indicate that the highest supported version is unbounded (or maybe follows semver, like caret in package-level specifications)?

**Comment by StefanKarpinski** [comment]
You're right, this should be documented. I will write up some docs at some point soon.

**Comment by timholy** [comment]
Would be nice to have this. There's a weird registry compatibility problem with Atom right now (https://discourse.julialang.org/t/juno-support-for-julia-1-2-0-not-ready-yet/27804) that also hitting me. (Seems like it should happen to anyone with JuliaInterpreter 0.7.0, see https://github.com/JunoLab/Atom.jl/pull/162). I was trying to fix it, but I've not dug into the internals enough to know the difference between "0.8.6-0", "0.7.0-*", or even the meaning of "0.6-0.7". Stuff like inclusive/exclusive bounds etc would be great to know. I can help write this if someone explains here.

**Comment by fredrikekre** [comment]
In short:
```
"0.8.6-0" # [0.8.6 - 1.0.0)
"0.7.0-*" # [0.7.0 - ∞)
"0.6-0.7" # [0.6.0 - 0.8.0)
```
You can use `VersionRange` to investigate, for example:
```julia
julia> import Pkg.Types: VersionRange

julia> v"0.8.5" ∈ VersionRange("0.8.6-0")
false

julia> v"0.8.6" ∈ VersionRange("0.8.6-0")
true

julia> v"0.9" ∈ VersionRange("0.8.6-0")
true

julia> v"1" ∈ VersionRange("0.8.6-0")
false
```
```julia
julia> import Pkg.Types: VersionRange

julia> v"0.8.5" ∈ VersionRange("0.8.6-*")
false

julia> v"0.8.6" ∈ VersionRange("0.8.6-*")
true

julia> v"1" ∈ VersionRange("0.8.6-*")
true

julia> v"1000" ∈ VersionRange("0.8.6-*")
true
```
```julia
julia> import Pkg.Types: VersionRange

julia> v"0.6.0" ∈ VersionRange("0.6-0.7")
true

julia> v"0.7.0" ∈ VersionRange("0.6-0.7")
true

julia> v"0.8" ∈ VersionRange("0.6-0.7")
false
```

**Comment by KristofferC** [comment]
Should be fixed by https://github.com/JuliaLang/Pkg.jl/pull/1349
